src/cmd/ksh93/sh/io.c

```diff
@@ -1969,6 +1969,14 @@ static ssize_t slowread(Sfio_t *iop,void *buff,register size_t size,Sfdisc_t *ha
 		errno = EINTR;
 		return(-1);
 	}
+	fno = sffileno(iop);
```

Comment:
Did you find a case where fno != 0?

Reply:
I didn't, but later on in the slowread function this isn't hardcoded as 0 either:
https://github.com/ksh93/ksh/blob/af298a350bf233ebf33128fe21771ee065a68083/src/cmd/ksh93/sh/io.c#L1986
Maybe a later commit could change this second sffileno(iop) to reuse fno, or replace both uses with just 0.

By the way, in the piperead function fd isn't reused here:
https://github.com/ksh93/ksh/blob/af298a350bf233ebf33128fe21771ee065a68083/src/cmd/ksh93/sh/io.c#L1919
https://github.com/ksh93/ksh/blob/af298a350bf233ebf33128fe21771ee065a68083/src/cmd/ksh93/sh/io.c#L1930